### ISSUE TITLE
Wasserstein GAN

### ISSUE THREAD

**Issue by leo-p** [leo-p/papers#16]
https://arxiv.org/pdf/1701.07875.pdf

We introduce a new algorithm named WGAN, an alternative to traditional GAN training. In this new model, we show that we can improve the stability of learning, get rid of problems like mode collapse, and provide meaningful learning curves useful for debugging and hyperparameter searches. Furthermore, we show that the corresponding optimization problem is sound, and provide extensive theoretical work highlighting the deep connections to other distances between distributions.

**Comment by leo-p** [comment]
## Summary:
- *Dataset:* [LSUN-Bedrooms](http://lsun.cs.princeton.edu/2016/)
- *Objective:* Robust unsupervised learning of a probability distribution using a new module called the `critic` and the `Earth-mover distance`.

## Inner working:
Basically train a `critic` until convergence to retrieve the Wasserstein-1 distance, see pseudo-algorithm below:

<img width="500" alt="screen shot 2017-05-03 at 5 05 09 pm" src="https://cloud.githubusercontent.com/assets/17261080/25667162/003c9330-3023-11e7-9081-c181011f4e6f.png">

## Results:
- Easier training: no need for batch normalization and no need to fine-tune generator/discriminator balance.
- Less sensitivity to network architecture.
- Very good proxy that correlates very well with sample quality.
- Non-vanishing gradients.
